geonode/settings.py

```diff
@@ -1956,56 +1955,64 @@ def get_geonode_catalogue_service():
-#    'allauth.socialaccount.providers.facebook',
-# )
+SOCIALACCOUNT_WITH_GEONODE_LOCAL_SINGUP = ast.literal_eval(
+    os.environ.get("SOCIALACCOUNT_WITH_GEONODE_LOCAL_SINGUP", "True")
+)
 
-SOCIALACCOUNT_PROVIDERS = {
-    "linkedin_oauth2": {
-        "SCOPE": [
-            "r_emailaddress",
-            "r_liteprofile",
-        ],
-        "PROFILE_FIELDS": [
-            "id",
-            "email-address",
-            "first-name",
-            "last-name",
-            "picture-url",
-            "public-profile-url",
-        ],
+# GeoNode Default Generic OIDC Provider
+SOCIALACCOUNT_OIDC_PROVIDER_ENABLED = ast.literal_eval(os.environ.get("SOCIALACCOUNT_OIDC_PROVIDER_ENABLED", "False"))
+SOCIALACCOUNT_OIDC_PROVIDER = os.environ.get("SOCIALACCOUNT_OIDC_PROVIDER", "geonode_openid_connect")
+SOCIALACCOUNT_ADAPTER = os.environ.get("SOCIALACCOUNT_ADAPTER", "geonode.people.adapters.GenericOpenIDConnectAdapter")
+
+# Enable this in order to enable the OIDC SocialAccount Provider
+if SOCIALACCOUNT_OIDC_PROVIDER_ENABLED:
+    INSTALLED_APPS += ("geonode.people.socialaccount.providers.geonode_openid_connect",)
+
+_AZURE_TENANT_ID = os.getenv("MICROSOFT_TENANT_ID", "")
+_AZURE_SOCIALACCOUNT_PROVIDER = {
+    "NAME": "Microsoft Azure",
+    "SCOPE": [
+        "User.Read",
+        "openid",
+    ],
+    "AUTH_PARAMS": {
+        "access_type": "online",
+        "prompt": "select_account",
     },
-    "facebook": {
-        "METHOD": "oauth2",
-        "SCOPE": [
-            "email",
-            "public_profile",
-        ],
-        "FIELDS": [
-            "id",
-            "email",
-            "name",
-            "first_name",
-            "last_name",
-            "verified",
-            "locale",
-            "timezone",
-            "link",
-            "gender",
-        ],
+    "COMMON_FIELDS": {"email": "mail", "last_name": "surname", "first_name": "givenName"},
+    "ACCOUNT_CLASS": "allauth.socialaccount.providers.azure.provider.AzureAccount",
+    "ACCESS_TOKEN_URL": f"https://login.microsoftonline.com/{_AZURE_TENANT_ID}/oauth2/v2.0/token",
+    "AUTHORIZE_URL": f"https://login.microsoftonline.com/{_AZURE_TENANT_ID}/oauth2/v2.0/authorize",
+    "PROFILE_URL": "https://graph.microsoft.com/v1.0/me",
+}
+
+_GOOGLE_SOCIALACCOUNT_PROVIDER = {
+    "NAME": "Google",
+    "SCOPE": [
+        "profile",
+        "email",
+    ],
+    "AUTH_PARAMS": {
+        "access_type": "online",
+        "prompt": "select_account consent",
     },
+    "COMMON_FIELDS": {"email": "email", "last_name": "family_name", "first_name": "given_name"},
+    "ACCOUNT_CLASS": "allauth.socialaccount.providers.google.provider.GoogleAccount",
+    "ACCESS_TOKEN_URL": "https://oauth2.googleapis.com/token",
+    "AUTHORIZE_URL": "https://accounts.google.com/o/oauth2/v2/auth",
+    "ID_TOKEN_ISSUER": "https://accounts.google.com",
+    "OAUTH_PKCE_ENABLED": True,
+}
+
+SOCIALACCOUNT_PROVIDERS = {
+    "geonode_openid_connect": _AZURE_SOCIALACCOUNT_PROVIDER,
```

Comment:
@afabiani why aren't we including also the `_GOOGLE_SOCIALACCOUNT_PROVIDER` here, if Azure is configured by default?

Reply:
@giohappy you need to choose one or another in this implementation. I rfactored this a bit so it is a bit more clear